System/Random.hs

```diff
@@ -1053,12 +1068,8 @@ class Random a where
   -- * For 'Integer', the range is (arbitrarily) the range of 'Int'.
   {-# INLINE random #-}
   random  :: RandomGen g => g -> (a, g)
-  random g = runGenState g genRandom
-
-  --{-# INLINE randomM #-}
-  randomM :: MonadRandom g s m => g s -> m a
-  -- default randomM :: (MonadRandom g m, Uniform a) => g -> m a
-  -- randomM = uniform
+  default random :: (RandomGen g, Uniform a) => g -> (a, g)
+  random g = runGenState g uniformM
```

Comment:
> Default implementation should be removed. It will break all custom instances of `Random` otherwise.

From https://github.com/idontgetoutmuch/random/pull/83#discussion_r408195766. Does this not apply here?

Reply:
Not quite. Before addition of `DefaultSignature` implementation the `random` function did not have any default implementation at all, so we are guaranteed  that all existing instances do provide it. In case of `randomM` that was a new function added to the class, so none of the current class had an implementation for it and would break if there was no `UnifromRange` instance. That was not a problem for all instances provided in the `random` library, but any custom instance for `Random` was guaranteed to break, because the could not possibly have had a `UniformRange` instance, because it was not available before.